File: Leetcode First 200 JS solution/26. Remove Duplicates from Sorted Array.js
var removeDuplicates = function(nums) {

    let k = 0;
    for (let i = 0; i < nums.length; i++) {
        if (!i || nums[i] != nums[i-1]){
            nums[k++] = nums[i];
        }
    }
    return k;
};

// o(n) solution

File: Daily Coding Practice/30 -day finish 300 coding challenge/2020-4-28. Two Pointers pattern/26. Remove Duplicates from Sorted Array.py
class Solution:
    def removeDuplicates(self, nums):
        nums[:] = sorted(set(nums))
        return len(nums)
# sort algorithm o(NlogN)
# space complexity o(N)

class Solution:
    def removeDuplicates(self, nums):
        # two pointer solution
        # nice solution !

        if len(nums) == 0: return 0
        i = 0
        for j in range(1, len(nums)):
            if nums[i] != nums[j]:
                i += 1
                nums[i] = arr[j]
        return i + 1

# time complexity o(N)
# space complexity o(1)
